base/broadcast.jl

```diff
@@ -216,36 +245,36 @@ function broadcast_t(f, ::Type{Any}, As...)
         return similar(Array{Union{}}, shape)
     end
     nargs = length(As)
-    keeps, Idefaults = map_newindexer(shape, As)
+    keeps, Idefaults = map_indexer(shape, As)
     st = start(iter)
     I, st = next(iter, st)
-    val = f([ As[i][newindex(I, keeps[i], Idefaults[i])] for i=1:nargs ]...)
+    val = f(( _broadcast_getindex(As[i], newindex(I, keeps[i], Idefaults[i])) for i=1:nargs )...)
```

Comment:
It also dispatches on a generator. I believe that's because the way this is parsed. Now, that you mentioned that could be a problem if people is trying to broadcast with `any` or `all`. But that should be another issue.